src/warm-cache-for-festival.ts

```diff
@@ -9,18 +9,26 @@ async function warm(festival: string, years: number[]) {
 
     for (const year of years) {
         const filename: string = festival + "_" + year + ".txt";
-        let file;
+        let   file;
+        let mtime;
         if (existsSync(filename)) {
             file = readFileSync(filename, "utf-8");
+            mtime = statSync(filename).mtime;
         } else if (existsSync(`lineups/${filename}`)) {
             file = readFileSync(`lineups/${filename}`, "utf-8");
+            mtime = statSync(`lineups/${filename}`).mtime;
         } else if (existsSync(`../lineups/${filename}`)) {
             file = readFileSync(`../lineups/${filename}`, "utf-8");
+            mtime = statSync(`../lineups/${filename}`).mtime;
         } else {
             console.error(`File ${filename} not found`);
             return;
         }
 
+        redisClient.set(`festival:${festival.toLowerCase()}_${year}:last_updated_date`,
```

Comment:
Can you move this to the very end of the `for (const year of years)` for loop instead of the beginning? That way it'll only set the updated date if all of the artists/songs are retrieved without a crash

Reply:
Done